### DISCUSSION TITLE
Does the code run differently under different OS?

### DISCUSSION THREAD

**Discussion by simonpark1113** [ANLAB-KAIST/KENSv3#39] [Project Q&A 1. Application Layer (Socket)]
Hi, I am very confused about what is happening right now.

In the beginning, my teammate was using macOS while I was using windows+WSL2.
After a while, we got the code to run almost flawlessly on my teammate's environment,
and only few debugging for few cases was left. 
At this point, he could perform communication between server and client using two terminals.

Then, after pulling the exact same code to my device, I ran the exact same commands,
but it did not work at all under my environment.
I thought the problem had to do with WSL2, so I installed actual Ubuntu 20.04 on my laptop,
but surprisingly, the code still would not run even under actual Ubuntu.

I have gone through numerous install&reinstall of environment and cmake&removing, so I think that I got everything else the same except for OS.

How could this be possible? Can identical codes run differently under different OS?
Or would it be that I must have missed some sort of setup?

**Comment by Pusnow** [comment]
Check your sources are *really identical*. Many students make mistakes while copying codes (check git pushed and pulled).

Then, try address sanitizer. Some memory errors can cause non-deteminism. https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Debugging-Memory-Bugs-(Only-for-macOS-and-Linux)

And please elaborate more, not just saying "not working".
